Modules/gcmodule.c

```diff
@@ -1940,6 +1940,88 @@ gc_get_freeze_count_impl(PyObject *module)
 }
 
 
+/*[clinic input]
+gc.is_immortal -> bool
+
+    instance: object
+        The instance to perform the immortal check on.
+
+Check if an object has been immortalized.
+[clinic start generated code]*/
+
+static int
+gc_is_immortal_impl(PyObject *module, PyObject *instance)
```

Comment:
I think is better if these two functions are only exported if `Py_IMMORTAL_INSTANCES` is defined (in the same way `sys.gettotalrefcount` is only exposed in debug mode. In this way you don't need to return a default "arbitrary" value if immortal objects are not activated.

Reply:
I had that but then the Windows builds started complaining that the clinic generated code was not being used anywhere. I don't think there's a way to add an ifdef in the clinic header.

Instead, I think I will just remove the clinic code and put the entire implementation here to ifdef guard all of it.

Reply:
> I had that but then the Windows builds started complaining that the clinic generated code was not being used anywhere. I don't think there's a way to add an ifdef in the clinic header.

If I recall correctly, the argument clinic should add it automatically if you add the `ifdef` around the definition it in the `.c` file

Reply:
This is the result after surrounding the code in the c file with `ifdef` and running `make clinic`:

```diff
diff --git a/Modules/clinic/gcmodule.c.h b/Modules/clinic/gcmodule.c.h
index 8742166199..9d0a9bead9 100644
--- a/Modules/clinic/gcmodule.c.h
+++ b/Modules/clinic/gcmodule.c.h
@@ -383,6 +383,8 @@ exit:
     return return_value;
 }

+#if defined(Py_IMMORTAL_INSTANCES)
+
 PyDoc_STRVAR(gc_is_immortal__doc__,
 "is_immortal($module, /, instance)\n"
 "--\n"
@@ -423,6 +425,10 @@ exit:
     return return_value;
 }

+#endif /* defined(Py_IMMORTAL_INSTANCES) */
+
+#if defined(Py_IMMORTAL_INSTANCES)
+
 PyDoc_STRVAR(gc_immortalize_heap__doc__,
 "immortalize_heap($module, /)\n"
 "--\n"
@@ -440,4 +446,14 @@ gc_immortalize_heap(PyObject *module, PyObject *Py_UNUSED(ignored))
 {
     return gc_immortalize_heap_impl(module);
 }
-/*[clinic end generated code: output=e017df03428c53c2 input=a9049054013a1b77]*/
+
+#endif /* defined(Py_IMMORTAL_INSTANCES) */
+
+#ifndef GC_IS_IMMORTAL_METHODDEF
+    #define GC_IS_IMMORTAL_METHODDEF
+#endif /* !defined(GC_IS_IMMORTAL_METHODDEF) */
+
+#ifndef GC_IMMORTALIZE_HEAP_METHODDEF
+    #define GC_IMMORTALIZE_HEAP_METHODDEF
+#endif /* !defined(GC_IMMORTALIZE_HEAP_METHODDEF) */
+/*[clinic end generated code: output=1dd9323a39790420 input=a9049054013a1b77]*/
diff --git a/Modules/gcmodule.c b/Modules/gcmodule.c
index 3c1954aa2f..17b9a1a781 100644
--- a/Modules/gcmodule.c
+++ b/Modules/gcmodule.c
@@ -1940,6 +1940,7 @@ gc_get_freeze_count_impl(PyObject *module)
 }


+#ifdef Py_IMMORTAL_INSTANCES
 /*[clinic input]
 gc.is_immortal -> bool

@@ -2020,6 +2021,7 @@ gc_immortalize_heap_impl(PyObject *module)
 #endif  /* Py_IMMORTAL_INSTANCES */
     Py_RETURN_NONE;
 }
+#endif


 PyDoc_STRVAR(gc__doc__,
```

Reply:
Oh sweet, let me try that!